File: ts/retrieve-linked-content/linked_content_get_article_with_author.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { createDeliveryClient } from '@kontent-ai/delivery-sdk';
import { Author } from './models/Author';
import { SimpleArticle } from './models/SimpleArticle';

const deliveryClient = createDeliveryClient({
  environmentId: '8d20758c-d74c-4f59-ae04-ee928c0816b7'
});

const response = await deliveryClient.item<SimpleArticle>('the_origin_of_coffee')
  .depthParameter(1)
  .toPromise();

const article: SimpleArticle = response.data.item;
const author: Author = article.elements.author;

File: java/retrieve-linked-content/linked_content_get_article_with_author.java
// Tip: Find more about Java SDK at https://kontent.ai/learn/java
import kontent.ai.delivery.*;

// Initializes a DeliveryClient
DeliveryClient client = new DeliveryClient("KONTENT_AI_ENVIRONMENT_ID");

// Tip: Create strongly typed models according to https://kontent.ai/learn/strongly-typed-models
// Registers the model class for simple articles
client.registerType(SimpleArticle.class);

CompletionStage<SimpleArticle> item = client.getItem(
	"the_origin_of_coffee",
	SimpleArticle.class,
	DeliveryParameterBuilder.params()
		.linkedItemsDepth(1)
		.build()
);
